src/duk_error_misc.c

```diff
@@ -42,6 +71,58 @@ DUK_INTERNAL duk_hobject *duk_error_prototype_from_code(duk_hthread *thr, duk_er
 DUK_INTERNAL void duk_err_setup_heap_ljstate(duk_hthread *thr, duk_small_int_t lj_type) {
 	duk_tval tv_tmp;
 
+#if defined(DUK_USE_DEBUGGER_SUPPORT)
+
+	/* If something is thrown with the debugger attached and nobody will
+	 * catch it, execution is paused before the longjmp, turning over
+	 * control to the debug client. This allows local state to be examined
+	 * before the stack is unwound.
+	 */
+
+	/* TODO: Allow customizing the pause and notify behavior */
+
+	if (DUK_HEAP_IS_DEBUGGER_ATTACHED(thr->heap) && lj_type == DUK_LJ_TYPE_THROW) {
+		duk_context *ctx = (duk_context *) thr;
+		duk_bool_t fatal;
+		duk_bool_t is_syntax_err;
+		duk_hobject *hobj;
+
+		/* Avoid intercepting syntax errors. Doing so will usually cause a
+		 * segfault since the JS executor isn't guaranteed to be active.
+		 */
+		is_syntax_err = duk_get_error_code(ctx, -1) == DUK_ERR_SYNTAX_ERROR;
```

Comment:
I guess the main issue for me is that I don't understand what happens when there's no executor running - and because that can happen with or without a SyntaxError, is there a concrete problem here?